permission_handler/ios/Classes/strategies/CriticalAlertsPermissionStrategy.m

```diff
@@ -0,0 +1,77 @@
+//
+//  CriticalAlertsPermissionStrategy.m
+//  permission_handler
+//
+//  Created by Neal Soni on 2021/6/8.
+//
+
+#import "CriticalAlertsPermissionStrategy.h"
+
+#if PERMISSION_CRITICAL_ALERTS
+
+@implementation CriticalAlertsPermissionStrategy
+
+- (PermissionStatus)checkPermissionStatus:(PermissionGroup)permission {
+  return [CriticalAlertsPermissionStrategy permissionStatus];
+}
+
+- (ServiceStatus)checkServiceStatus:(PermissionGroup)permission {
+  return ServiceStatusNotApplicable;
+}
+
+- (void)requestPermission:(PermissionGroup)permission completionHandler:(PermissionStatusHandler)completionHandler {
+  PermissionStatus status = [self checkPermissionStatus:permission];
+  if (status != PermissionStatusDenied) {
+    completionHandler(status);
+    return;
+  }
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
What is the reason to dispatch this whole block on the main queue?

Reply:
Ngl, I just stripped down the `NotificationPermissionStrategy.m` and replace the `authorizationOptions` to be the `UNAuthorizationOptionCriticalAlert`. I assumed there was a reason for doing it this way for notifications so didn't change anything.

https://github.com/Baseflow/flutter-permission-handler/blob/274247d0f980538008eb36d7b202197343fc60f1/permission_handler/ios/Classes/strategies/NotificationPermissionStrategy.m#L28-L30

Reply:
Alright, thank you for the response. As mentioned I will do a sweep over all strategies on a later point and make sure these settings are setup correctly.